### PR TITLE
Added endpoint-config template

### LINKED ISSUE #3637: Complete ZAP Templates
#### Problem
Two files are not generated with ZAP
endpoint_config.h
client-command-macro.h

 #### Proposed Solution
Create templates for those files

### PR DESCRIPTION
**Pull request by jepenven-silabs**
#### Problem
Some required ZCL templates are missing
 #### Summary of Changes
 Added the `endpoint_config.h` templates for zap generation.

 Partial Fixes #3637

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]
@vivien-apple Please take a look?

**Review by saurabhst — APPROVED** [review]

**Review by rwalker-apple — APPROVED** [review]

**Review by mspang — APPROVED** [review]

File: src/app/zap-templates/endpoint_config.zapt [hunk 1]
```diff
@@ -3,4 +3,100 @@
...
+
+// Array of EmberAfPluginReportingEntry structures.
+#define ZRD(x) EMBER_ZCL_REPORTING_DIRECTION_ ## x
+#define ZAP_REPORT_DIRECTION(x) ZRD(x)
+
+#define EMBER_AF_GENERATED_REPORTING_CONFIG_DEFAULTS_TABLE_SIZE ({{endpoint_reporting_config_default_count}})
+#define EMBER_AF_GENERATED_REPORTING_CONFIG_DEFAULTS {{endpoint_reporting_config_defaults}}
+
+
+{{/endpoint_config}}
```
Comment (vivien-apple) [inline src/app/zap-templates/endpoint_config.zapt:102]: All the `#define EMBER_AF_GENERATED_PLUGIN_*` code seems to be gone.

I understand the issue to generate those, but we need to find a solution, we can not just remove them.

File: src/app/zap-templates/endpoint_config.zapt [hunk 2]
```diff
@@ -3,4 +3,100 @@
...
+// This is an array of EmberAfManufacturerCodeEntry structures for attributes.
+#define GENERATED_ATTRIBUTE_MANUFACTURER_CODE_COUNT ({{endpoint_attribute_manufacturer_code_count}})
+#define GENERATED_ATTRIBUTE_MANUFACTURER_CODES {{endpoint_attribute_manufacturer_codes}}
+
+// Array of EmberAfPluginReportingEntry structures.
+#define ZRD(x) EMBER_ZCL_REPORTING_DIRECTION_ ## x
+#define ZAP_REPORT_DIRECTION(x) ZRD(x)
+
+#define EMBER_AF_GENERATED_REPORTING_CONFIG_DEFAULTS_TABLE_SIZE ({{endpoint_reporting_config_default_count}})
+#define EMBER_AF_GENERATED_REPORTING_CONFIG_DEFAULTS {{endpoint_reporting_config_defaults}}
```
Comment (vivien-apple) [inline src/app/zap-templates/endpoint_config.zapt:99]: {{r.minOrSource}} uses to be **1** previously. Seems to be **0**. Do you know why ?

**Review by vivien-apple — CHANGES_REQUESTED** [review]

File: src/app/zap-templates/endpoint_config.zapt [hunk 3]
```diff
@@ -3,4 +3,100 @@
...
+#define ZAP_MIN_MAX_DEFAULTS_INDEX(index) {(uint8_t*)(&minMaxDefault[index])}
+
+// This is an array of EmberAfAttributeMinMaxValue structures.
+#define GENERATED_MIN_MAX_DEFAULT_COUNT {{endpoint_attribute_min_max_count}}
+#define GENERATED_MIN_MAX_DEFAULTS {{endpoint_attribute_min_max_list}}
+
+#define ZAP_ATTRIBUTE_MASK(mask) ATTRIBUTE_MASK_ ## mask
+// This is an array of EmberAfAttributeMetadata structures.
+#define GENERATED_ATTRIBUTE_COUNT {{endpoint_attribute_count}}
+#define GENERATED_ATTRIBUTES {{ endpoint_attribute_list }}
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:30]: Weird things are compiling just fine both for the EFR32 demo and the `all-cluster-app` on my side.

File: src/app/zap-templates/endpoint_config.zapt (hunk 1 above)
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:102]: @vivien-apple Could EMBER_AF_GENERATED_PLUGIN_* be replace with a generated cluster define instead ? Something like CHIP_GENERATED_CLUSTER_*cluster_name* ? Or is there a use case where this solution wouldn't work ?

File: src/app/zap-templates/endpoint_config.zapt (hunk 3 above)
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:30]: Done

**Comment by bzbarsky-apple** [comment]
@jepenven-silabs Could you rebase to tip to resolve the merge conflict and hopefully the CI failures?

**Comment by jepenven-silabs** [comment]
> @jepenven-silabs Could you rebase to tip to resolve the merge conflict and hopefully the CI failures?

@bzbarsky-apple currently working on it to fix @vivien-apple comments, I'll do the rebase once those comments are resolved

File: src/app/zap-templates/endpoint_config.zapt (hunk 3 above)
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:30]: Fixed

[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:30]: This should be file as a different issue since it's xml specific.

File: src/app/zap-templates/endpoint_config.zapt [hunk 4]
```diff
@@ -3,4 +3,100 @@
...
+
+// This is an array of EmberAfCluster structures.
+#define ZAP_ATTRIBUTE_INDEX(index) ((EmberAfAttributeMetadata *)(&generatedAttributes[index]))
+
+#define GENERATED_CLUSTER_COUNT {{endpoint_cluster_count}}
+#define GENERATED_CLUSTERS {{endpoint_cluster_list}}
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:36]: Fixed

[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:36]: @vivien-apple in that case a new issue should be open to fix this

File: src/app/zap-templates/endpoint_config.zapt (hunk 2 above)
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:99]: it's 0 for the min value of the attribute.

[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:99]: The brace problem is resolved. As for the `chip-zcl-pro-codec-api.h` this seems to be a problem specific to that templates as the `included` field is parsed and used within zap. A separate issue should be open to address this. On that note, I noticed that this templates use mostly custom helper which override native ZAP helper, so maybe something was lost in there..

File: src/app/zap-templates/endpoint_config.zapt (hunk 1 above)
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:113]: @vivien-apple a temporary solution was implemented. We should discuss about a more permanent one after the Holidays in a software TT meeting.

File: src/app/zap-templates/endpoint_config.zapt (hunk 2 above)
[the remark this answers is not in the document]
Comment (vivien-apple) [reply src/app/zap-templates/endpoint_config.zapt:111]: No. The issue is that `endpoint_config` (https://github.com/project-chip/zap/blob/b1320c815ae633108c14f6b601115447c18333f5/src-electron/generator/helper-endpointconfig.js#L543) does not use the `included` field of the `.zap` configuration file. 

As an example https://github.com/project-chip/connectedhomeip/blob/master/examples/all-clusters-app/all-clusters-common/all-clusters-app.zap#L1148 is marked as `"included": 0` but it appears in the `GENERATED_ATTRIBUTES` and the `EMBER_AF_GENERATED_REPORTING_CONFIG_DEFAULTS` of the new `gen/endpoint_config.h`.

Likely because at the end of the day `endpoint_config` does not use the `ENDPOINT_TYPE_ATTRIBUTE` table but uses a custom mechanism to build the list of attributes.

That's likely the root of the big jump in `ATTRIBUTE_LARGEST` too. From *8* to *254*

File: src/app/zap-templates/endpoint_config.zapt (hunk 3 above)
[the remark this answers is not in the document]
Comment (vivien-apple) [reply src/app/zap-templates/endpoint_config.zapt:32]: Can you open a followup and add the reference here ? It is probably fine right now but that's really something we need to fix.

File: src/app/zap-templates/endpoint_config.zapt (hunk 1 above)
Comment (vivien-apple) [reply src/app/zap-templates/endpoint_config.zapt:113]: Any reasons to not uses the macros from `gen_config.h` to check for the plugins in `util.cpp` instead of adding new macros in `endpoint_config.h` ?

Comment (vivien-apple) [reply src/app/zap-templates/endpoint_config.zapt:113]: A more specific example. Can you use `EMBER_AF_PLUGIN_TEMPERATURE_MEASUREMENT_SERVER` instead of `CHIP_TEMP_MEASUREMENT_CLUSTER_SERVER` ?

File: src/app/zap-templates/endpoint_config.zapt (hunk 4 above)
[the remark this answers is not in the document]
Comment (vivien-apple) [reply src/app/zap-templates/endpoint_config.zapt:39]: It does not sounds right to just remove methods clusters relies on to works properly.
What is the issue to generate those exactly ? I assume that's the same root issue than plugins ?

File: src/app/zap-templates/endpoint_config.zapt [hunk 5]
```diff
@@ -3,4 +3,117 @@
...
+#define FIXED_NETWORKS {{endpoint_fixed_network_array}}
+
+
+// Array of EmberAfCommandMetadata structs.
+#define ZAP_COMMAND_MASK(mask) COMMAND_MASK_ ## mask
+#define EMBER_AF_GENERATED_COMMAND_COUNT ({{endpoint_command_count}})
+#define GENERATED_COMMANDS {{endpoint_command_list}}
+
+// Array of EmberAfManufacturerCodeEntry structures for commands.
+#define GENERATED_COMMAND_MANUFACTURER_CODE_COUNT ({{endpoint_command_manufacturer_code_count}})
```
Comment (vivien-apple) [inline src/app/zap-templates/endpoint_config.zapt:86]: I'm pretty sure it does not work. I have recently added a `MfgSpecificPing` method to chip, and generating this file with it does not increment this number.

File: src/app/zap-templates/endpoint_config.zapt [hunk 6]
```diff
@@ -3,4 +3,117 @@
...
+#define GENERATED_COMMAND_MANUFACTURER_CODES {{endpoint_command_manufacturer_codes}}
```
Comment (vivien-apple) [inline src/app/zap-templates/endpoint_config.zapt:87]: Same issue as above.

File: src/app/zap-templates/endpoint_config.zapt (hunk 2 above)
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:111]: Thanks for the precision and Good catch, I'll fix this asap

[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:111]: @vivien-apple as for the Largest attribute, this is because the COLOR CONTROL cluster has the attribute compensation text enabled (which is a string up to 254 characters)

File: src/app/zap-templates/endpoint_config.zapt (hunk 3 above)
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:32]: Done issue #4272

File: src/app/zap-templates/endpoint_config.zapt (hunk 5 above)
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:86]: Fixed

File: src/app/zap-templates/endpoint_config.zapt (hunk 6 above)
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:87]: Fixed

File: src/app/zap-templates/endpoint_config.zapt (hunk 4 above)
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:39]: Long story short these methods and array are silabs specific and have no direct link with the ZCL part. Therefore, ZAP cannot generate them. However, we have a SDK extensions feature in ZAP that we could use to generate those arrays and link them. So either we modify the zcl framework to fit this new reality or we use the sdk extension to generate the `GENERATED_FUNCTIONS_ARRAYS` either way it should be done in a separate issue because of the amount of work required.

[the remark this answers is not in the document]
Comment (vivien-apple) [reply src/app/zap-templates/endpoint_config.zapt:39]: > Long story short these methods and array are silabs specific and have no direct link with the ZCL part. Therefore, ZAP cannot generate them. However, we have a SDK extensions feature in ZAP that we could use to generate those arrays and link them. So either we modify the zcl framework to fit this new reality or we use the sdk extension to generate the `GENERATED_FUNCTIONS_ARRAYS` either way it should be done in a separate issue because of the amount of work required.

The issue is that https://github.com/project-chip/connectedhomeip/blob/515d2ccc76736eff344e5fd2acd0ef25a7863cbc/examples/lighting-app/nrfconnect/main/ZclCallbacks.cpp#L87 depends on such mechanism.

Maybe that's something we need to fix first. Do you mind pointing me to the "SDK extensions feature" that you mentioned ?

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap [hunk 7]
```diff
@@ -44,24 +44,6 @@
...
-            },
-            {
-              "name": "MfgSpecificPing",
-              "code": 0,
-              "mfgCode": "0x1002",
-              "source": "client",
-              "incoming": 1,
-              "outgoing": 1
-            }
-          ],
```
Comment (vivien-apple) [inline examples/all-clusters-app/all-clusters-common/all-clusters-app.zap:64]: Can you rebase ? We should keep the "MfgSpecificPing" command here. That should be easier to use it now that #4192 has landed.

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap [hunk 8]
```diff
@@ -160,23 +169,6 @@
...
-              "included": 1,
-              "storageOption": "RAM",
-              "singleton": 0,
-              "bounded": 0,
-              "defaultValue": "2",
-              "reportable": 0,
-              "minInterval": 0,
-              "maxInterval": 65344,
-              "reportableChange": 0
-            }
```
Comment (vivien-apple) [inline examples/all-clusters-app/all-clusters-common/all-clusters-app.zap:179]: Not part of this PR but it would be nice for the ZAP code that generate the files to not move around the content of the file when it has not changed. I have not looked at it at all. One possibility for such behavior could be that it uses a JS object {} to store some values and then iterate over it. But those objects are not ordered. Or it could be something else.

File: src/app/zap-templates/endpoint_config.zapt (hunk 4 above)
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt:39]: Here's the documentation about the SDK. https://github.com/project-chip/zap/blob/master/docs/sdk-integration.md

[the remark this answers is not in the document]
Comment (vivien-apple) [reply src/app/zap-templates/endpoint_config.zapt:39]: > Here's the documentation about the SDK. https://github.com/project-chip/zap/blob/master/docs/sdk-integration.md

Do you have a direct link to an example of the SDK feature that may help us to address this use case ? It would make my life easier instead of having to found something in the whole SDK documentation :)

[the remark this answers is not in the document]
Comment (vivien-apple) [reply src/app/zap-templates/endpoint_config.zapt:39]: I do not have a good answer yet in order to solve the mentioned issue. What do you think of hardcoding this part in the template at the moment and fill a followup ? It will let us move forward.

[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/endpoint_config.zapt]: A partly Hard-coded solution is now implemented to generate the `GENERATED_FUNCTIONS` macro

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap (hunk 8 above)
Comment (jepenven-silabs) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.zap:179]: The ZAP team is aware of this issue and it's in their pipeline to fix this.

File: examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h [hunk 9]
```diff
@@ -15,365 +15,392 @@
...
-#define ATTRIBUTE_LARGEST (8)
+#define ATTRIBUTE_LARGEST (254)
+
 // Total size of singleton attributes
-#define ATTRIBUTE_SINGLETONS_SIZE (6)
+#define ATTRIBUTE_SINGLETONS_SIZE (0)
 
 // Total size of attribute storage
-#define ATTRIBUTE_MAX_SIZE 150
+#define ATTRIBUTE_MAX_SIZE (422)
```
Comment (vivien-apple) [inline examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h:382]: It still looks weird that `ATTRIBUTE_MAX_SIZE' goes to `422`. I think you already answered me but I can't find your answer anymore. Can you explain it to me again ? (sorry).

File: examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h [hunk 10]
```diff
@@ -391,232 +418,351 @@
...
-
-#define EMBER_AF_GENERATED_PLUGIN_STACK_STATUS_FUNCTION_CALLS                                                                      \
-    emberAfPluginReportingStackStatusCallback(status);                                                                             \
-    emberAfPluginTemperatureMeasurementServerStackStatusCallback(status);                                                          \
-    emberAfPluginIasZoneServerStackStatusCallback(status);
-
-// Generated data for the command discovery
+// Array of EmberAfCommandMetadata structs.
+#define ZAP_COMMAND_MASK(mask) COMMAND_MASK_##mask
+#define EMBER_AF_GENERATED_COMMAND_COUNT (165)
```
Comment (vivien-apple) [inline examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h:426]: Do you know why we are going from `161`to `165` ?

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap [hunk 11]
```diff
@@ -19,23 +19,23 @@
...
-      "path": "../../../../third_party/zap/repo/zcl-builtin/silabs/zcl.json",
+      "path": "../../../../../zap/zcl-builtin/silabs/zcl.json",
       "version": "ZCL Test Data",
       "type": "zcl-properties"
     },
     {
       "pathRelativity": "relativeToZap",
-      "path": "../../../../src/app/zap-templates/chip-templates.json",
-      "version": "chip-v1",
+      "path": "../../../../../zap/test/gen-template/zigbee/gen-templates.json",
```
Comment (vivien-apple) [inline examples/all-clusters-app/all-clusters-common/all-clusters-app.zap:28]: This changes is strange.

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap [hunk 12]
```diff
@@ -5614,7 +5606,7 @@
       "endpointTypeName": "Anonymous Endpoint Type",
       "endpointTypeIndex": 1,
       "profileId": "0x0103",
-      "endpointId": 1,
+      "endpointId": 2,
```
Comment (vivien-apple) [inline examples/all-clusters-app/all-clusters-common/all-clusters-app.zap:5609]: Why ?

File: examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h [hunk 13]
```diff
@@ -391,232 +418,351 @@
...
+            {                                                                                                                      \
+                ZAP_REPORT_DIRECTION(REPORTED), 0x0001, 0x0300, 0x0007, ZAP_CLUSTER_MASK(SERVER), 0x0000, { { 0, 65344, 0 } }      \
+            }, /* Reporting for cluster: "Color Control", attribute: "color temperature". side: server */                          \
+            {                                                                                                                      \
+                ZAP_REPORT_DIRECTION(REPORTED), 0x0001, 0x0402, 0x0000, ZAP_CLUSTER_MASK(SERVER), 0x0000, { { 0, 65344, 0 } }      \
+            }, /* Reporting for cluster: "Temperature Measurement", attribute: "measured value". side: server */                   \
+            {                                                                                                                      \
+                ZAP_REPORT_DIRECTION(REPORTED), 0x0002, 0x0006, 0x0000, ZAP_CLUSTER_MASK(SERVER), 0x0000, { { 0, 65344, 0 } }      \
```
Comment (vivien-apple) [inline examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h:766]: It seems like all the `{ 1, 65534, 0}` values has been changed to `{ 0, 65534, 0}`. Do you know why ?

**Review by vivien-apple — CHANGES_REQUESTED** [review]
Very close. Just a few additional questions/nits.

File: examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h (hunk 9 above)
Comment (vivien-apple) [reply examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h:382]: Ah right. I remember it cames mostly "from compensation text". That said I don't remember why it is added ?

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap (hunk 12 above)
Comment (jepenven-silabs) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.zap:5609]: Must be because the two endpoint had the same id of 1. ZAP doesn't auto-increment endpoint id when adding a new one.

File: examples/temperature-measurement-app/esp32/main/gen/endpoint_config.h [hunk 14]
```diff
@@ -14,117 +14,110 @@
...
+        { ZAP_CLUSTER_INDEX(0), 2, 12 },                                                                                           \
     }
-#define GENERATED_ATTRIBUTE_MANUFACTURER_CODE_COUNT (0)
 
 // Largest attribute size is needed for various buffers
 #define ATTRIBUTE_LARGEST (2)
+
 // Total size of singleton attributes
-#define ATTRIBUTE_SINGLETONS_SIZE (6)
+#define ATTRIBUTE_SINGLETONS_SIZE (0)
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply examples/temperature-measurement-app/esp32/main/gen/endpoint_config.h:96]: Files within `/zcl-builtin/silabs/general.xml` can change at the leasure of silabs. If we want to modify them we should have our own copy of chip xml files stored somewhere in our repo so we could modify them freely. Also I think this is outside of the PR scope.

File: examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h (hunk 9 above)
Comment (jepenven-silabs) [reply examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h:382]: It is because the attribute is enabled in the configuration and the maximum string length of said attribute is 256 ish.

File: src/app/zap-templates/templates/app/endpoint_config.zapt [hunk 15]
```diff
@@ -3,4 +3,114 @@
...
+#define GENERATED_ATTRIBUTE_MANUFACTURER_CODE_COUNT ({{endpoint_attribute_manufacturer_code_count}})
+#define GENERATED_ATTRIBUTE_MANUFACTURER_CODES {{endpoint_attribute_manufacturer_codes}}
+
+// Array of EmberAfPluginReportingEntry structures.
+#define ZRD(x) EMBER_ZCL_REPORTING_DIRECTION_ ## x
+#define ZAP_REPORT_DIRECTION(x) ZRD(x)
+
+{{#if (endpoint_reporting_config_default_count)}}
+// TODO Issue #3871 Reporting should only be enabled if there are reportable attributes
+// Use this macro to check if Reporting plugin is included
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/templates/app/endpoint_config.zapt:106]: yes it does

File: examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h (hunk 13 above)
Comment (jepenven-silabs) [reply examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h:766]: minOrSource is set to 0 in the database.

File: examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h [hunk 16]
```diff
@@ -391,232 +418,351 @@
...
-            { 0x0B03, 0x01, COMMAND_MASK_INCOMING_CLIENT }, /* Appliance Statistics / LogResponse */                               \
-            { 0x0B03, 0x02, COMMAND_MASK_INCOMING_CLIENT }, /* Appliance Statistics / LogQueueResponse */                          \
-            { 0x0B03, 0x03, COMMAND_MASK_INCOMING_CLIENT }, /* Appliance Statistics / StatisticsAvailable */                       \
+        { 0x0000, 0x00,                                                                                                            \
+          ZAP_COMMAND_MASK(INCOMING_CLIENT) | ZAP_COMMAND_MASK(OUTGOING_CLIENT) }, /* Basic (client): ResetToFactoryDefaults */    \
+            {                                                                                                                      \
+                0x0000, 0x00, ZAP_COMMAND_MASK(INCOMING_SERVER) | ZAP_COMMAND_MASK(OUTGOING_SERVER)                                \
+            }, /* Basic (server): ResetToFactoryDefaults */                                                                        \
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h:433]: Fixed

File: examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h (hunk 10 above)
Comment (jepenven-silabs) [reply examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h:426]: From memory it is because the commands between endpoint are not regrouped anymore with zap. Which leads to the duplication of commands for the second endpoint with the On/Off cluster thus increasing the Command Count.

File: src/app/zap-templates/templates/app/helper.js [hunk 17]
```diff
@@ -186,6 +187,133 @@ function asChipUnderlyingType(label, type)
...
+// these helpers are a Hot fix for the "GENERATED_FUNCTIONS" problem
+// They should be removed or replace once issue #4369 is resolved
+// These helpers only works within the endpoint_config iterator
+
+// List of all cluster with generated functions
+var endpointClusterWithInit = [ 'Identify', 'Groups', 'Scenes', 'On/off', 'Level Control', 'Color Control', 'IAS Zone' ];
+var endpointClusterWithAttributeChanged = [ 'Identify', 'Door Lock' ];
+var endpointClusterWithPreAttribute     = [ 'IAS Zone' ];
+var endpointClusterWithMessageSent      = [ 'IAS Zone' ];
+
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply src/app/zap-templates/templates/app/helper.js:200]: Looks good but won't work unless `clusterGeneratedFunctions` includes all possible cluster. Otherwise you'll get cannot access property length of undefined in the `if (clusterGeneratedFunctions[clusterName].length)`. I would keep it as it is since it is a hot fix and this will be improve later on.

File: examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h [hunk 18]
```diff
@@ -15,365 +15,392 @@
...
-    const EmberAfGenericClusterFunction emberAfFuncArrayIdentifyClusterServer[] = {                                                \
+    const EmberAfGenericClusterFunction chipFuncArrayIdentifyServer[] = {                                                          \
         (EmberAfGenericClusterFunction) emberAfIdentifyClusterServerInitCallback,                                                  \
-        (EmberAfGenericClusterFunction) emberAfIdentifyClusterServerAttributeChangedCallback                                       \
+        (EmberAfGenericClusterFunction) emberAfIdentifyClusterServerAttributeChangedCallback,                                      \
+    };                                                                                                                             \
+    const EmberAfGenericClusterFunction chipFuncArrayGroupsServer[] = {                                                            \
+        (EmberAfGenericClusterFunction) emberAfGroupsClusterServerInitCallback,                                                    \
```
[the remark this answers is not in the document]
Comment (jepenven-silabs) [reply examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h:256]: Yeah it's because of the current implementation of the hot fix. Will be fixed in another PR.

File: examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h (hunk 13 above)
Comment (vivien-apple) [reply examples/all-clusters-app/all-clusters-common/gen/endpoint_config.h:768]: Ah yeah. I feel like I already asked that... Sorry.

File: examples/all-clusters-app/all-clusters-common/all-clusters-app.zap [hunk 19]
```diff
@@ -5605,16 +5605,16 @@
   "endpoints": [
     {
       "endpointTypeName": "Anonymous Endpoint Type",
-      "endpointTypeIndex": 0,
+      "endpointTypeIndex": 1,
       "profileId": "0x0103",
       "endpointId": 1,
       "networkId": 1
```
Comment (vivien-apple) [inline examples/all-clusters-app/all-clusters-common/all-clusters-app.zap:5611]: Please change all `networkId` occurrences to `0` inside all `*.zap` files. This is likely the source of the `Cirque` failure. as I guess this is what prevent `Cirque` from running.

That's not your fault since `networkId: 1` is here since a long time, but generating `endpoint_id.h` instead of the old version from `AppBuilder` has exposed the issue.

I guess it brings us back to #1937 for the root issue.

I will approve the PR with this change ! :)

Comment (jepenven-silabs) [reply examples/all-clusters-app/all-clusters-common/all-clusters-app.zap]: Done !

**Review by vivien-apple — APPROVED** [review]
Thanks ! 

Let's see if `Cirque` is happier now.
